### DISCUSSION TITLE
Inter-comunication by cable and BLE

### DISCUSSION THREAD

**Discussion by Salva-OA** [pybricks/support#492] [Q&A] [answered]
Hi

Based on this info: 
https://www.philohome.com/wedo2reverse/protocol.htm
https://antonsmindstorms.com/product/distance-sensor-breakout-board-for-spike-prime-and-robot-inventor/
http://www.mindsensors.com/spike-prime/218-breadboard-connector-kit-for-spike-prime-ultrasonic-sensor-cable
https://code.pybricks.com/static/docs/iodevices/pupdevice.html

Can we have a cable connection between two hub ports, swapping TX by RX pins like on a serial rs232 link? 
Can we have a library (class, methods, properties, ...) that may hide the level of communication to just a function to define a port as a hardware extension port?
and then just listen to the ASCII strings that we might receive from, plus a function to send ASCII strings to the port?
This would be also the method to change UART into wifi or cabled rj-45 external boards to expand the hub to other higher systems.

Following the same idea, Can you make a new class that would enable the BLE link to a remote device like a Microbit gamepad, where could make a "pairing", and later for this the same functions to control ASCII strings sent or received?
For this BLE intercommunication, it would be nice to have an external tool to identify the address of booth devices and fix it later con the code of the hub, and restrict messages one to one.

If possible please update the web info:
https://code.pybricks.com/static/docs/tools/index.html

Thanks for your help
Salva

**Comment by dlech** [comment, marked as answer]
> Can we have a cable connection between two hub ports, swapping TX by RX pins like on a serial rs232 link?
Can we have a library (class, methods, properties, ...) that may hide the level of communication to just a function to define a port as a hardware extension port?
and then just listen to the ASCII strings that we might receive from, plus a function to send ASCII strings to the port?
This would be also the method to change UART into wifi or cabled rj-45 external boards to expand the hub to other higher systems.


Yes, please give https://github.com/pybricks/support/issues/220 a :+1: if you would like to see this feature implemented.



> Following the same idea, Can you make a new class that would enable the BLE link to a remote device like a Microbit gamepad, where could make a "pairing", and later for this the same functions to control ASCII strings sent or received?
> For this BLE intercommunication, it would be nice to have an external tool to identify the address of booth devices and fix it later con the code of the hub, and restrict messages one to one.

Yes, please see https://github.com/pybricks/support/issues/262 for more info. Feel free to add comments there if you have any uses cases that haven't been covered already.

**Reply by Salva-OA** [reply]
Thanks a lot, David
